lib/edgehog_device_forwarder/termination_callbacks.ex

```diff
@@ -0,0 +1,62 @@
+# Copyright 2023 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+defmodule EdgehogDeviceForwarder.TerminationCallbacks do
+  use GenServer
+
+  @table :termination_callbacks_table
+
+  @spec add(pid, (-> any)) :: :ok
+  def add(pid, on_close) when is_pid(pid) and is_function(on_close, 0) do
+    GenServer.cast(__MODULE__, {:add, pid, on_close})
+  end
+
+  @spec start_link(any) :: GenServer.on_start()
+  def start_link(opts \\ []) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  def cache_spec do
+    params = [name: @table, ttl_check_interval: false]
+    id = {ConCache, @table}
+
+    {ConCache, params}
+    |> Supervisor.child_spec(id: id)
+  end
+
+  @impl GenServer
+  def init(_opts) do
+    # If the server is restarted, our pid changed
+    #   hence, to receive the :DOWN, we need to re-monitor existing processes
+    #   Process.monitor also sends the :DOWN for already dead pid, so we don't need to check that.
+    ConCache.ets(@table)
+    |> :ets.tab2list()
+    |> Enum.map(fn {pid, _} -> pid end)
+    |> Enum.each(&Process.monitor/1)
```

Comment:
This seems a good fit for a for comprehension:
```suggestion
    table = ConCache.ets(@table)
    
    for {pid, _ } <- :ets.tab2list(table) do
      Process.monitor(pid)
    end
```